src/yz_solrq.erl

```diff
@@ -18,800 +18,241 @@
-            end;
-        false ->
-            gen_server:reply(From, ok),
-            State2
-    end.
+%% @doc Return the status of all solrq workers.
+-spec status() -> [{atom(), yz_solrq_worker:status()}].
+status() ->
+    [{Name, yz_solrq_worker:status(Name)} || Name <- solrq_worker_names()].
+
+%% @doc Return the list of solrq names registered with this supervisor
+-spec solrq_worker_names() -> [atom()].
+solrq_worker_names() ->
+    tuple_to_list(get_solrq_worker_tuple()).
+
+%% @doc Return the list of solrq names registered with this supervisor
+-spec solrq_helper_names() -> [atom()].
+solrq_helper_names() ->
+    tuple_to_list(get_solrq_helper_tuple()).
+
+%% @doc return the queue capacity as a percentage in [0..100], representing the ratio of
+%% data stored in data in all the queues to the potential capacity of all the queues.
+-spec queue_capacity() -> capacity().
+queue_capacity() ->
+    TotalQueueLength = lists:sum([yz_solrq_worker:all_queue_len(Name) || Name <- tuple_to_list(get_solrq_worker_tuple())]),
+    TotalCapacity = lists:sum([yz_solrq_worker:get_hwm(Name) || Name <- tuple_to_list(get_solrq_worker_tuple())]),
+    round(100 * (TotalQueueLength / TotalCapacity)).
 
-%% @doc Enqueue the entry and return updated state.
-enqueue(E, #indexq{queue = Q, queue_len = L, aux_queue = A, draining = Draining} = IndexQ) ->
-    case Draining of
-        false ->
-            IndexQ#indexq{queue = queue:in(E, Q), queue_len = L + 1};
-        _ ->
-            IndexQ#indexq{aux_queue = queue:in(E, A)}
-    end.
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
 
-%% @doc Re-enqueue undelivered items as part of updated indexq.
-requeue_undelivered(Undelivered, #indexq{queue = Queue, queue_len = QueueLen} = IndexQ) ->
-    NewQueue = queue:join(queue:from_list(Undelivered), Queue),
-    IndexQ#indexq{
-      queue = NewQueue,
-      queue_len = QueueLen + erlang:length(Undelivered)
-     }.
-
-%% @doc Trigger a flush and return state
-flush(Index, IndexQ, State) ->
-    IndexQ2 = request_worker(Index, IndexQ),
-    update_indexq(Index, IndexQ2, State).
-
-%% @doc handle a blown fuse by setting the fuse_blown flag,
-%%      purging any data if required, and unblocking any vnodes
-%%      if required.
-handle_blown_fuse(Index, #state{purge_strategy=PBIStrategy} = State) ->
-    IndexQ = get_indexq(Index, State),
-    State1 = update_indexq(Index, IndexQ#indexq{fuse_blown = true}, State),
-    State2 = maybe_purge([{Index, IndexQ}], PBIStrategy, State1),
-    maybe_unblock_vnodes(State2).
-
-%% @doc purge entries depending on purge strategy if we are over the HWM
-maybe_purge(Indices, PBIStrategy, State) ->
-    case over_hwm(State) of
-        true ->
-            purge(Indices, PBIStrategy, State);
-        _ ->
-            State
+-spec child_count(atom()) -> non_neg_integer().
+child_count(ChildType) ->
```

Comment:
child_count and do_child_resize won't work as-is because they are using ?MODULE, which used to be yz_solrq_sup. Thinking the best thing to do is to move the implementation of these back to yz_solrq_sup, and keep these functions here but just pass through to the supervisor.